hack/run-k8s-e2e.sh

```diff
@@ -0,0 +1,136 @@
+
+_kubectl=cluster-up/cluster-up/kubectl.sh
+_virtctl=./hack/tools/bin/virtctl
+_default_clusterctl_path=./hack/tools/bin/clusterctl
+_default_tmp_path=./hack/tools/bin/tmp
+
+export CLUSTERCTL_PATH=${CLUSTERCTL_PATH:-${_default_clusterctl_path}}
+
+function ensure_cluster_up {
+    ${_kubectl} get ns || ret=$?
+    echo "Return $ret"
+    if [ -z "$ret" ]; then
+	echo "Cluster running"
+    else
+        echo "Cluster not running, starting cluster"
+	make cluster-up
+    fi
+}
+
+function ensure_synced {
+    # This appears to not work if run multiple times. The sync fails if run a second time.
+    make cluster-sync-split
+}
+
+function get_control_plane_vm_name {
+    vms_list=$(${_kubectl} get vm -n ${TENANT_CLUSTER_NAMESPACE} --no-headers -o custom-columns=":metadata.name")
+    for vm in $vms_list
+    do
+        if [[ "$vm" == ${TENANT_CLUSTER_NAME}-control-plane* ]]; then
+            control_plane_vm_name=$vm
+	    return 0
+        fi
+    done
+    echo "control-plane vm is not found in namespace ${TENANT_CLUSTER_NAMESPACE} (looking for regex ${TENANT_CLUSTER_NAME}-control-plane*)"
+    exit 1
+}
+
+function start_tenant_api_forward {
+    # Use the infra cluster kubeconfig from the environment variable otherwise the forward will fail
+    echo "${_virtctl} port-forward -n $TENANT_CLUSTER_NAMESPACE vmi/$control_plane_vm_name 64443:6443"
+    ${_virtctl} port-forward -n $TENANT_CLUSTER_NAMESPACE vmi/$control_plane_vm_name 64443:6443 > /dev/null 2>&1 &
+    trap 'kill $(jobs -p) > /dev/null 2>&1' EXIT
+}
+
+function get_tenant_kubeconfig_for_e2e {
+   echo "$CLUSTERCTL_PATH get kubeconfig ${TENANT_CLUSTER_NAME} -n ${TENANT_CLUSTER_NAMESPACE} > .${TENANT_CLUSTER_NAME}-kubeconfig-e2e"
+   $CLUSTERCTL_PATH get kubeconfig ${TENANT_CLUSTER_NAME} -n ${TENANT_CLUSTER_NAMESPACE} > .${TENANT_CLUSTER_NAME}-kubeconfig-e2e
+   # Modify the kubeconfig to achieve 3 things
+   # 1. Change API ip address to 127.0.0.1:64443 (the port forward)
+   # 2. Add insecure-skip-tls-verify: true
+   # 3. Remove the CA which cannot be defined if insecure-skip-tls-verify: true is set
+   # Need the insecure because the port-forward has changed the ip address and the cert doesn't know about 127.0.0.1
+   sed -i 's/server:.*/server: https:\/\/127.0.0.1:64443/' .${TENANT_CLUSTER_NAME}-kubeconfig-e2e
+   sed -i '/server:.*/a \ \ \ \ insecure-skip-tls-verify: true' .${TENANT_CLUSTER_NAME}-kubeconfig-e2e
+   sed -i '/certificate-authority-data/d' .${TENANT_CLUSTER_NAME}-kubeconfig-e2e
+   tenant_kubeconfig=.${TENANT_CLUSTER_NAME}-kubeconfig-e2e
+}
+
+function ensure_e2e_binary {
+    if [ ! -f "e2e.test" ]; then
+	# Would prefer to detect k8s version from tenant cluster.
+        curl --location https://dl.k8s.io/v1.22.0/kubernetes-test-linux-amd64.tar.gz |   tar --strip-components=3 -zxf - kubernetes/test/bin/e2e.test kubernetes/test/bin/ginkgo
+    else
+	echo "Binary exists"
+    fi
+}
+
+function enable_sshuttle {
```

Comment:
maybe we could run all this in a pod within the infra cluster to avoid sshuttle usage?!

example of how conformance tests run in capk today.
https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/03d6f013b1309fada53ffd3276532f52f8b03199/hack/run-conformance.sh#L83

Reply:
Good idea, I updated the script to do that.